lib/src/layout/layout.dart

```diff
@@ -69,33 +76,177 @@ class _AgoraVideoViewerState extends State<AgoraVideoViewer> {
+            crossAxisAlignment: CrossAxisAlignment.center,
+            children: [
+              Text(
+                widget
+                    .client.sessionController.value.connectionData!.channelName
+                    .toString(),
+                style: TextStyle(
+                  color: Colors.white,
+                  fontSize: 24,
+                ),
+                textAlign: TextAlign.center,
+              ),
+              SizedBox(
+                width: 10,
+              ),
+              GestureDetector(
+                onTap: () {
+                  String tempTokenString = widget.client.sessionController.value
+                              .connectionData!.tempToken ==
+                          null
+                      ? ''
+                      : '&tempToken=${widget.client.sessionController.value.connectionData!.tempToken}';
+                  String tokenUrlString = widget.client.sessionController.value
+                              .connectionData!.tokenUrl ==
+                          null
+                      ? ''
+                      : '&tokenUrl=${widget.client.sessionController.value.connectionData!.tokenUrl}';
+                  Share.share(
+                      'Please join my channel: "${widget.client.sessionController.value.connectionData!.channelName}" using this link: ${widget.client.deepLinkBaseUrl}/channelName=${widget.client.sessionController.value.connectionData!.channelName}' +
+                          tempTokenString +
+                          tokenUrlString);
+                },
+                child: Icon(
+                  Icons.copy,
+                  color: Colors.white,
+                  size: 24,
+                ),
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    late String appId;
+    late String channelName;
+    String? tempToken;
+    String? tokenUrl;
+    // TODO: List<AreaCode>? areaCodes;
+    String? baseUrl;
+    String? layoutType = "grid";
+    Future<void> updateAgoraConnectionDataConfig(Object? link) async {
+      if (link != '') {
+        StreamSubscription<Uri?> subscription =
+            uriLinkStream.listen((Uri? uri) async {
+          print('uri: $uri');
+          baseUrl = uri?.origin;
+
+          uri?.queryParametersAll.forEach((key, value) {
+            print('params');
+            print('$key: ${value[0]}');
+            if (key == 'channelName') {
+              channelName = value[0];
+              print('Channe Name: $channelName');
+            } else if (key == 'tempToken') {
+              tempToken = value[0];
+              print('Temp Token: $tempToken');
+            } else if (key == 'tokenUrl') {
+              tokenUrl = value[0];
+              print('Token Url: $tokenUrl');
+            } else if (key == "layoutTpype") {
+              layoutType = value[0];
+              print("layoutType: $layoutType");
+            }
```

Comment:
A switch statement would be better here, + a case to catch and print an unknown key

Reply:
if adding a default case to the switch case, you can remove the first two prints (lines 192 + 193)